stores/migrations/mysql/main/migration_00002_prune_slabs_trigger.sql

```diff
@@ -0,0 +1,52 @@
+-- prune manually before creating trigger
+DELETE slabs
+FROM slabs
+LEFT JOIN slices ON slices.db_slab_id = slabs.id
+WHERE slices.db_object_id IS NULL
+AND slices.db_multipart_part_id IS NULL
+AND slabs.db_buffered_slab_id IS NULL;
+
+-- remove ON DELETE CASCADE from slices
```

Comment:
What's the reason for replacing these cascades with triggers?

Reply:
A CASCADE deletion doesn't activate triggers in MySQL. So the deleted object would delete the associated slices but the slices wouldn't trigger the slab pruning trigger.

[source](https://dev.mysql.com/doc/refman/8.3/en/create-trigger.html)